lib/parser/char-buffer.cc

```diff
@@ -53,4 +53,42 @@ void CharBuffer::Put(const char *data, std::size_t n) {
 
 void CharBuffer::Put(const std::string &str) { Put(str.data(), str.size()); }
 
+std::string CharBuffer::Marshal() const {
+  std::string result;
+  std::size_t bytes{bytes_};
+  result.reserve(bytes);
+  for (const Block &block : blocks_) {
+    std::size_t chunk{std::min(bytes, Block::capacity)};
+    for (std::size_t j{0}; j < chunk; ++j) {
+      result += block.data[j];
+    }
+    bytes -= chunk;
+  }
+  result.shrink_to_fit();
```

Comment:
Is shrink_to_fit going to do anything after having already reserved the right capacity?

Reply:
Probably not, but it can't hurt in that case.  It's present here to release any overallocation that `reserve` may have thought was prudent for growth, and as a signal to the reader that the string is complete at this point.